functorch/_src/compilers.py

```diff
@@ -20,6 +20,12 @@ def _canonicalize(fx_g):
     return fx_g
 
 
+def _prepare_for_jit_script(gm):
```

Comment:
Can we rename this to something like `strip_overloads`, add a docstring, and also add a `gm.recompile()`?

Reply:
added